library/core/src/ptr/const_ptr.rs

```diff
@@ -23,11 +28,12 @@ impl<T: ?Sized> *const T {
     /// assert!(!ptr.is_null());
     /// ```
     #[stable(feature = "rust1", since = "1.0.0")]
+    #[rustc_const_unstable(feature = "const_ptr_is_null", issue = "74939")]
     #[inline]
-    pub fn is_null(self) -> bool {
+    pub const fn is_null(self) -> bool {
         // Compare via a cast to a thin pointer, so fat pointers are only
         // considering their "data" part for null-ness.
-        (self as *const u8) == null()
+        (self as *const u8).guaranteed_eq(null())
```

Comment:
Is there an obvious reason why this is a better implementation than `!guaranteed_ne`?

Reply:
nope, do you prefer the latter?

Reply:
I have no idea which one makes more sense. Let's land it the way you wrote, but could you write a comment in the tracking issue that explains how these two different implementations differ in behavior?

Reply:
The version based on `guaranteed_ne` is effectively a "maybe_null" test. It returns `true` for all out-of-bounds pointers.

Reply:
I mentioned this in the issue